### ISSUE TITLE
false positive type error for associative used as ifn

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#479]
```
$ clj-kondo --lint - <<< '(remove (assoc {} :a 1) [:a])'
<stdin>:1:9: error: Expected: function, received: associative collection.
```

**Comment by p-himik** [comment]
I still see this false positive on sets with clj-kondo 2019.10.26:
```bash
$ clj-kondo --lint - <<< '(filter (conj #{} 1) [1])'
<stdin>:1:9: error: Expected: function, received: collection.
```

**Comment by borkdude** [comment]
@p-himik Thanks for reporting, I made a new issue for it:

https://github.com/borkdude/clj-kondo/issues/554
